src/transactions/transactionBuilder/transactionBuilder.ts

```diff
@@ -445,6 +451,11 @@ export function generateSignedTransactionForSimulation(args: InputSimulateTransa
 }
 
 export function getAuthenticatorForSimulation(publicKey: PublicKey) {
+  if (publicKey instanceof KeylessPublicKey || publicKey instanceof Secp256k1PublicKey) {
+    // eslint-disable-next-line no-param-reassign
+    publicKey = new AnyPublicKey(publicKey);
+  }
```

Comment:
I think ideally we should make it so that `KeylessPublicKey` or `Secp256k1PublicKey` never make it here.
One of the reasons for having a different interface for `PublicKey` and `AccountPublicKey` 

We can figure this out later

Reply:
Yeah, agreed and noted.